### ISSUE TITLE
Engine should store temporary files in /tmp

### ISSUE THREAD

**Issue by basimr** [project-lovelace/lovelace-engine#22]
This will reduce file pollution in the repo's directory.
The Engine should also delete temporary files it's done with.

**Comment by ali-ramadhan** [comment]
Actually I think this was a bad idea that gave us some really weird issues.

I can't find where I wrote down the details, but essentially I think the issue was that we were pushing the file into `/tmp/` so quick after startup that Ubuntu was cleaning `/tmp/` and deleting it as part of its usual boot process. This led to issues where we definitely pushed the code file, but couldn't find it anywhere in the VM.

So I ended up using `/root/` And in general, it seems like a file moved into `/tmp/` isn't guaranteed to be there always, the OS may decide to clean it up?

But I agree we should find a better place. Maybe `/root/tmp/`?
